src/typecheck/mod.rs

```diff
@@ -1314,6 +1314,52 @@ fn row_add(
     }
 }
 
+/// Get the set of all unification variables in a type.
+fn get_free_variables(wildcards: &Vec<TypeWrapper>, t: TypeWrapper) -> HashSet<usize> {
+    match t {
+        TypeWrapper::Ptr(i) => HashSet::from([i]),
+        TypeWrapper::Concrete(ty) => match ty {
+            AbsType::Dyn()
+            | AbsType::Num()
+            | AbsType::Bool()
+            | AbsType::Str()
+            | AbsType::Sym()
+            | AbsType::Flat(_)
+            | AbsType::Var(_)
+            | AbsType::RowEmpty() => HashSet::new(),
+            // TODO: replace with get_wilccard_var(..)
```

Comment:
Is there a reason this hasn't been done as part of this PR? From what I can tell, calling `get_wildcard_var` here could change the behaviour of this function (by returning any new unification variables created), but I'm struggling to work out whether that would have any impact on `occurs_check`.

Reply:
I think both version are correct, because `get_wildcard_vars` may only create new fresh unification variables, which don't matter (as they will never fail the occurs check). Somehow the current implementation is lazier, as it doesn't generate the wildcard vars yet, but will the next time they are actually needed by unification. It probably doesn't make any noticeable difference, and if anything, I expect this version to be slightly better, as it doesn't pollute the hashset with new fresh unification variables.